### DISCUSSION TITLE
SKR-E3-DIP + S42B on XY + TCM2209 ZE

### DISCUSSION THREAD

**Discussion by 4slash1977** [Jyers/Marlin#952] [Q&A] [answered]
Hi - can any one tell me if Jyers Extensible UI can be used with SKR-E3-DIP 1.2 board and I would like to use S42B v2 Closed Loop drivers for X,Y and TCM2209 for Z,E (Z will be dual stepper in series)?

**Comment by Jyers** [comment, marked as answer]
That should work fine if you compile your own firmware. Just a note: the display cable may need to be re-pinned to work correctly as I believe the SKR boards use a different pin layout

**Reply by 4slash1977** [reply]
I used Ender 3 V2 Extensible UI - with BLTouch-3x3 template - I change:
default_envs = STM32F103RE_btt
#define MOTHERBOARD BOARD_BTT_SKR_E3_DIP

I can't compile this, is that a right path for firmware or I'm geting this wrong?

**Reply by Jyers** [reply]
What errors are you getting? There might be other options needed for that board, I'd recommend looking online

**Reply by 4slash1977** [reply]
I got:
*** [.pio\build\STM32F103RE_btt\src\src\HAL\STM32F1\eeprom_bl24cxx.cpp.o] Error 1
*** [.pio\build\STM32F103RE_btt\src\src\HAL\STM32F1\Servo.cpp.o] Error 1
*** [.pio\build\STM32F103RE_btt\src\src\HAL\STM32F1\SPI.cpp.o] Error 1
*** [.pio\build\STM32F103RE_btt\src\src\HAL\STM32F1\HAL_SPI.cpp.o] Error 1
*** [.pio\build\STM32F103RE_btt\src\src\HAL\STM32F1\eeprom_flash.cpp.o] Error 1
*** [.pio\build\STM32F103RE_btt\src\src\HAL\STM32F1\dogm\u8g_com_stm32duino_swspi.cpp.o] Error 1

**Reply by Jyers** [reply]
Interesting. I can't really make out much of the issue from that, have you tried compiling with a config meant for that board to make sure that at least works?

**Reply by 4slash1977** [reply]
Ok so that is due to V2 display - as long as I start with any template for E3 V1, PRO, MAX and I change chip type, serial and board type I can compile firware for BTT E3 DIP - but when I try it with template for V2 it dosn't compile. I have to look if anybody resolved this in general.

**Reply by Jyers** [reply]
Just checking you are starting with one of my V2 templates right? Not the ones from the main marlin repo

**Reply by 4slash1977** [reply]
Same thing with Your template and from main Marlin. As long as I start with V2 it dont compile for this board.

**Reply by 4slash1977** [reply]
So at this point I have some progres - I have to add to pins_BTT_SKR_E3_DIP.h :
#if ENABLED(DWIN_CREALITY_LCD)
  // RET6 DWIN ENCODER LCD
  #define BTN_ENC                           PB5
  #define BTN_EN1                           PB15
  #define BTN_EN2                           PB8
  #ifndef BEEPER_PIN
    #define BEEPER_PIN                      PA15
    #undef SPEAKER
  #endif
#endif

And that works for Marlin 2.0.7 - but when I want to compile same config with marlin >= 2.0.8 I have problem with multi serial configuration - it should be:
#define SERIAL_PORT 2
#define SERIAL_PORT_2 -1
But that only works in 2.0.7
2.0.8 - will compile when I coment out second serial port.

Do You have any advice - I know that from 2.0.8 something changed in Marlin in regards to serial

**Reply by Jyers** [reply]
If it compiled when commenting out the second serial port what is the issue? It looks like you're setting that port to -1 anyway so it shouldn't matter right? Or does that board need the other serial port defined for some reason?

**Reply by 4slash1977** [reply]
So that is out of my understanding :-) I'm not sure what serial port set to -1 mean. And what is second serial port for. I'm connecting from octoprint via usb so that is serial connection. I'm not sure what is second connection for - if it is for LCD - then I'm not sure how creality board works with only one serial set.

**Reply by Jyers** [reply]
I believe the serial port for the V2 LCD is actually setup somewhere in the dwin code rather than in the configs. Try it without defining the second port and see if that works for you.

**Reply by 4slash1977** [reply]
Finally I made it work :-) Changes I made (only those that are necessary):
PLATFORMIO.INI
default_envs = STM32F103RE_btt
CONFIGURATION.H
#define SERIAL_PORT -1
//#define SERIAL_PORT_2 2
#define MOTHERBOARD BOARD_BTT_SKR_E3_DIP
CONDITIONALS_LCD.H
#if ENABLED(DWIN_CREALITY_LCD)
  #define HAS_DISPLAY 1
  #define SERIAL_CATCHALL 0
  #ifndef LCD_SERIAL_PORT
    #define LCD_SERIAL_PORT 1 // this is wher I defined serial port for LCD
  #endif
#endif
PINS_BTT_SKR_E3_DIP.H
/**
 *        Ender 3 V2 display                         SKR E3 DIP 1.1
 *                _____                                     _____
 *            5V | 1 2 | GND                          5V   | 1 2 | GND
 *   (BTN_E1) A  | 3 4 | B (BTN_E2)         (BTN_EN1) PB7  | 3 4 | PB8   (BTN_E2)
 *          BEEP | 5 6   ENT (BTN_ENC)       (BEEPER) PB9  | 5 6   PA10  (BTN_ENC)
 *  (SKR_RX1) TX | 7 8 | RX (SKR_TX1)                 RST  | 7 8 | PA9
 *            NC | 9 10| NC                           PB6  | 9 10| PA15
 *                -----                                     -----
 *                EXP1                                      EXP1
 
 *    * Needs custom cable:
 *
 *   Display       Adapter   Board
 *                _________
 *   1 - 5V      ----------- 1 - 5V
 *   2 - GND     ----------- 2 - GND
 *   3 - BNT_E1  ----------- 3 - PB7
 *   4 - BNT_E2  ----------- 4 - PB8
 *   5 - BEEP    ----------- 5 - PB9
 *   6 - BNT_ENC ----------- 9 - PB6
 *   7 - TX      ----------- 6 - RX
 *   8 - RX      ----------- 8 - TX
 *   9 - NC      ----------- 7 - RST
 *  10 - NC      ----------- 10 - PA15
 */
#if ENABLED(DWIN_CREALITY_LCD)

  // RET6 DWIN ENCODER LCD
  #define BTN_ENC                           PB6
  #define BTN_EN1                           PB7
  #define BTN_EN2                           PB8

  #ifndef BEEPER_PIN
    #define BEEPER_PIN                      PB9
    #undef SPEAKER
  #endif
#endif

Those changes are necessary to make everything compile and work - I used Your Ender_3_V2_Extensible_UI with BLTouch-3x3-HighSpeed - configuration template. Original display work's fine and also Serial connection via USB is working.
